pkg/agent/agent.go

```diff
@@ -1132,26 +1135,17 @@ func getRoundInfo(bridgeClient ovsconfig.OVSBridgeClient) types.RoundInfo {
 	return roundInfo
 }
 
-func (i *Initializer) getNodeMTU(transportInterface *net.Interface) (int, error) {
+func (i *Initializer) getInterfaceMTU(transportInterface *net.Interface) (int, error) {
 	if i.mtu != 0 {
 		return i.mtu, nil
 	}
-	mtu := transportInterface.MTU
+	mtu := transportInterface.MTU - i.networkConfig.MTUDeduction
 	// Make sure mtu is set on the interface.
 	if mtu <= 0 {
 		return 0, fmt.Errorf("Failed to fetch Node MTU : %v", mtu)
 	}
-	if i.networkConfig.TrafficEncapMode.SupportsEncap() {
-		if i.networkConfig.TunnelType == ovsconfig.VXLANTunnel {
-			mtu -= config.VXLANOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GeneveTunnel {
-			mtu -= config.GeneveOverhead
-		} else if i.networkConfig.TunnelType == ovsconfig.GRETunnel {
-			mtu -= config.GREOverhead
-		}
-		if i.nodeConfig.NodeIPv6Addr != nil {
-			mtu -= config.IPv6ExtraOverhead
-		}
+	if i.nodeConfig.NodeIPv6Addr != nil {
+		mtu -= config.IPv6ExtraOverhead
```

Comment:
I wonder what should we do for Mutlicluster GW MTU. Do we support IPv6 tunnel for Multicluster GW? If so, we should include this into NetworkConfig.MTUDeduction. Even we do not, we should track this. Maybe at least add a comment on NetworkConfig.MTUDeduction to say it counts only IPv4 tunnel overhead, no IPv6, IPsec, and WireGuard overhead.

Reply:
We don't support IPv6 tunnel for MC GW. comment is added.